go/test/endtoend/vreplication/fk_test.go

```diff
@@ -102,6 +102,51 @@ func TestFKWorkflow(t *testing.T) {
 	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Running.String())
 	targetKs := vc.Cells[cellName].Keyspaces[targetKeyspace]
 	targetTab := targetKs.Shards["0"].Tablets[fmt.Sprintf("%s-%d", cellName, targetTabletId)].Vttablet
+
+	// Stop the LoadSimulator while we are testing for workflow error, so that
+	// we don't error out in the LoadSimulator as we will be shutting down source dbServer.
+	if withLoad {
+		cancel()
+		<-ch
+	}
+
+	sourceTab := vc.Cells[cellName].Keyspaces[sourceKeyspace].Shards["0"].Tablets[fmt.Sprintf("%s-%d", cellName, 100)]
+
+	// Stop the source database server to simulate an error during replication phase
+	// This should cause recoverable errors that atomic workflows should retry
+	// as it is already out of copy phase.
+	err := sourceTab.DbServer.Stop()
+	require.NoError(t, err)
+
+	// Give some time for the workflow to encounter errors and potentially retry
+	time.Sleep(5 * time.Second)
+
+	// Verify workflow is still running and hasn't terminated due to errors
+	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Running.String())
+
+	// Restart the source database to allow workflow to continue
+	err = sourceTab.DbServer.StartProvideInit(false)
+	require.NoError(t, err)
+
+	time.Sleep(2 * time.Second)
```

Comment:
Let's have a retry loop here, otherwise the next call might error out if the database takes time to restart resulting in a flaky test.